doc/make_tables.py

```diff
@@ -0,0 +1,246 @@
+import os
+
+import pyvista as pv
+
+CHARTS_TABLE_DIR = "api/plotting/charts"
+CHARTS_IMAGE_DIR = "images/charts"
+COLORS_TABLE_DIR = "api/utilities"
+
+
+class DocTable:
+    """Helper class to create tables for the documentation."""
+
+    path = None  # Path to the rst file to which the table will be written
+
+    @classmethod
+    def generate(cls):
+        assert cls.path is not None, "Subclasses should specify a path."
+        if os.path.exists(cls.path):
+            os.remove(cls.path)
+        tokens = cls.fetch_tokens()
+        with open(cls.path, "w", encoding="utf-8") as f:
+            f.write(cls.get_header(tokens))
+            for i, (token, data) in enumerate(tokens.items()):
+                row = cls.get_row(i, token, data)
+                if row is not None:
+                    f.write(row)
+        pv.close_all()
+
+    @classmethod
+    def fetch_tokens(cls):
+        raise NotImplementedError("Subclasses should specify a fetch_tokens method.")
+
+    @classmethod
+    def get_header(cls, tokens):
+        raise NotImplementedError("Subclasses should specify a table header.")
+
+    @classmethod
+    def get_row(cls, i, token, data):
+        raise NotImplementedError("Subclasses should specify a get_row method.")
+
+
+class LineStyleTable(DocTable):
+    """Class to generate line style table."""
+
+    path = f"{CHARTS_TABLE_DIR}/pen_line_styles.rst"
+    header = """
+.. list-table:: Line styles
+   :widths: 20 40 40
+   :header-rows: 1
+
+   * - Style
+     - Description
+     - Example
+"""
```

Comment:
There's a release in the very near future, so prettifying refactors like this can be a follow up (or just discarded).

Reply:
Can you check the colors table? I had to use `dedent` and `indent` to make it work properly. If you prefer that approach, I can quickly update the other tables as well.

Reply:
Well, I personally hate it when indented strings leak out to the global indentation level :) So I do prefer the current version despite the need to `indent`. But I'm not sure how others feel about this, this is pretty subjective.

Reply:
Personally I like the indentation of the new version (as you said no leaking to higher indentation levels), but the extra `indent` call feels a bit weird :/

Reply:
Hmm, too bad we have 3-space indentation here. Otherwise we could mimic [the dedent example](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) with
```py
    row_template = """
    * - {}
      - ``{}``
      - .. raw:: html
           <span style='width:100%; height:100%; display:block; background-color: {};'>&nbsp;</span>
    """
```
and this would keep Python's 4-space indentation as a literal part of the string. Which I guess is similar to what you originally had here.

Reply:
What is your opinion of something like this:

```py
    row_template = textwrap.dedent(
        """
        |   * - {}
        |     - ``{}``
        |     - .. raw:: html
        |
        |          <span style='width:100%; height:100%; display:block; background-color: {};'>&nbsp;</span>
        """
    ).replace('|', '')
```

A bit hacky, I admit, but it has proper indentation, works, and looks rather clean. We could also make it a separate method (e.g. `aligned_dedent` to give it a name), such that we don't have the `replace('|','')` at the end every time.

Reply:
I don't hate it :) Arguably every solution will be hacky or ugly, and this is indeed tidy and only a bit hacky. I think you can go for it. If you define that helper, I suggest making it `_aligned_dedent` and thus "private".